### PR TITLE
Pooled Staking Runtime API

### PR DESCRIPTION
**Pull request by nanocryk**
Stake in currency amount is not available in storage, and instead must be computed from the amount of shares and the value of each share. The pending manual rewards also needs to be computed based on the difference between the pool counter and delegator "checkpoint" times the amount of shares.

To simplify the implementation of UIs, a new Runtime API is added to perform those computations.

### CODE REVIEW

**Comment by tmpolaczyk** [comment]
You need to update api-augment:

```
cd typescript-api
pnpm tsx ./scripts/generate-types.ts
```

**Comment by girazoki** [comment]
I would suggest we use the same procedure as in the CI. Compile the node and do:

```
cd typescript-api
pnpm install
pnpm create-local-interfaces
```

File: pallets/pooled-staking/src/lib.rs [hunk 1]
```diff
@@ -517,4 +495,39 @@ pub mod pallet {
             Calls::<T>::update_candidate_position(&candidates)
         }
     }
+
+    impl<T: Config> Pallet<T> {
+        pub fn staked_in_pool(
+            candidate: Candidate<T>,
+            delegator: Delegator<T>,
+            pool: AllTargetPool,
+        ) -> Option<T::Balance> {
```
Comment (tmpolaczyk) [inline pallets/pooled-staking/src/lib.rs:504]: What does `None` mean here? If the stake is 0 this should return `Some(0)`, no?

**Comment by tmpolaczyk** [comment]
Also can you add some integration tests?

**Pull request closed without merging**